test/runner/worker.js

```diff
@@ -72,7 +73,9 @@ async function gracefullyCloseAndExit() {
   // Force exit after 30 seconds.
   setTimeout(() => process.exit(0), 30000);
   // Meanwhile, try to gracefully close all browsers.
-  await gracefullyCloseAll();
+  if (testRunner)
+    await testRunner.stop();
+  await fixturePool.teardownScope('worker');
```

Comment:
Should we do this concurrently to `testRunner.stop()`? I am afraid that slow test won't resolve `stop()` in 30 seconds. If we close the browser immediately, the test will fail on the next playwright operation - this seems safer.